py-polars/src/functions/meta.rs

```diff
@@ -56,3 +56,21 @@ pub fn get_float_fmt() -> PyResult<String> {
     };
     Ok(strfmt.to_string())
 }
+
+#[pyfunction]
+pub fn set_float_precision(precision: u8) -> PyResult<()> {
+    use polars_core::fmt::set_float_precision;
+    if precision > 16 && precision != u8::MAX {
```

Comment:
Again, why is the maximum supported precision 16 digits?

Reply:
@alexander-beedie That really depends on what you're doing. Every floating point number has an *exact* decimal representation (the converse isn't true), and someone might care about that. Most often people just go for "round-trip correctness", but this leads to silly things as this

```rust
println!("{}", 1u64 << 63);
println!("{}", (1u64 << 63) as f64);
```

printing

```
9223372036854775808
9223372036854776000
```

Note that `9223372036854775808` *is* the exact value. The overemphasis to round-trip correctness has also lead to real-world data loss. Rust (and `serde`) by default
print `f32`s with less precision because "only a few digits is needed for a correct-roundtrip". The problem is that reading back the values as `f64` gives you *different* values than if you were to cast `f32` as `f64` (which is completely lossless).

So if you print `f64` values with only as few digits as needed to round-trip it is correct *if read back in as an f64*, but if you were to read it out as, say, `Decimal`, you would get **wrong values**.

```python
>>> Decimal(math.pi) == Decimal(str(math.pi))
False
```

Reply:
You may want to read this part of a blog post I made that goes into more detail: https://orlp.net/blog/ordering-numbers/#floating-point-numbers-are-exact.